web/pages/api/save-image.ts

```diff
@@ -0,0 +1,38 @@
+// pages/api/saveImage.js
+import { NextApiRequest, NextApiResponse } from "next";
+import { db } from "../../db/firebase"; // Firestore database
+import { doc, setDoc, Timestamp } from "firebase/firestore";
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method === "POST") {
```

Comment:
API isn't needed in this case since Firebase can do client-side database calls, so to speed up code fetching we should call the Firebase functions as endpoints rather than in the API. There are some cases where we need to handle things on the server-side, but we're good in this case